protocol/storage/miner.go

```diff
@@ -806,7 +806,9 @@ func (sm *Miner) submitPoSt(start, end *types.BlockHeight, seed proofs.PoStChall
 		commRs[i] = input.commR
 	}
 
-	proofs, faults, err := sm.generatePoSt(commRs, seed)
+	sortedCommRSlice := types.NewSortedCommRSlice(commRs...)
+
+	proofs, faults, err := sm.generatePoSt(sortedCommRSlice.Values(), seed)
```

Comment:
BLOCKING: `generatePoSt` should be modified to accept a value of type `SortedCommRSlice`, and the `GeneratePoStRequest` should be modified to look like this:

```
type GeneratePoStRequest struct {
	CommRs        SortedCommRSlice
	ChallengeSeed proofs.PoStChallengeSeed
}
```

Similarly, the `VerifyPoStRequest` type should be modified to look like:

```
type VerifyPoSTRequest struct {
	ChallengeSeed PoStChallengeSeed
	CommRs        SortedCommRSlice
	Faults        []uint64
	Proofs        []PoStProof
	StoreType     SectorStoreType // used to control sealing/verification performance
}
```

Rationale: The type enforces an invariant, which is that these CommRs are sorted identically in both PoSt generation and verification flows. This type should to make its way as close to the CGO call as possible to ensure that we're enforcing the invariant.